### PR TITLE
QML: disable networking

### LINKED ISSUE #2902: [security] Disable networking from QML?
We should consider disabling network access from QML (eg via [setNetworkAccessible](https://doc.qt.io/qt-5/qnetworkaccessmanager.html#setNetworkAccessible)) if possible.

RE: https://twitter.com/SarahJamieLewis/status/1260323514130694145

qml-remote

Although I have no indication this could be _remotely_ exploited, it's a defense-in-depth ("belt and suspenders") opportunity.

### PR DESCRIPTION
**Pull request by selsta**
Resolves #2902

### CODE REVIEW

File: src/qt/NetworkAccessBlockingFactory.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    BlockedNetworkAccessManager(QObject *parent)
+        : QNetworkAccessManager(parent)
+    {
+        setProxy(QNetworkProxy(QNetworkProxy::Socks5Proxy, QLatin1String("0.0.0.0"), 0));
+    }
+
+protected:
+    virtual QNetworkReply *createRequest(Operation op, const QNetworkRequest &req, QIODevice *outgoingData = 0)
+    {
+        qWarning() << "QML attempted to load a network resource from" << req.url() << " - this is potentially an input sanitization flaw.";
```
Comment (xiphon) [inline src/qt/NetworkAccessBlockingFactory.h:55]: let's do it qCritical

File: src/qt/NetworkAccessBlockingFactory.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ */
+#include <QDebug>
+
+class BlockedNetworkAccessManager : public QNetworkAccessManager
+{
+public:
+    BlockedNetworkAccessManager(QObject *parent)
+        : QNetworkAccessManager(parent)
+    {
+        setProxy(QNetworkProxy(QNetworkProxy::Socks5Proxy, QLatin1String("0.0.0.0"), 0));
```
Comment (xiphon) [inline src/qt/NetworkAccessBlockingFactory.h:49]: Looking at the docs `setNetworkAccessible` is not marked as deprecated. Could you please double-check this?

Comment (selsta) [reply src/qt/NetworkAccessBlockingFactory.h:49]: https://doc-snapshots.qt.io/qt5-5.15/qnetworkaccessmanager-obsolete.html#setNetworkAccessible

File: src/qt/NetworkAccessBlockingFactory.h (hunk 1 above)
Comment (selsta) [reply src/qt/NetworkAccessBlockingFactory.h:55]: updated

**Comment by selsta** [comment]
I could also add a `#if` to set `setNetworkAccessible` on < Qt 5.15 and else `setProxy`.

**Pull request merged**